mico-admin/src/app/app-detail-public-ip/app-detail-public-ip.component.ts

```diff
@@ -77,11 +77,13 @@ export class AppDetailPublicIpComponent implements OnInit, OnChanges, OnDestroy
                     application.services.forEach(service => {
 
                         safeUnsubscribeList(this.subServiceInterfaces);
+                        this.subServiceInterfaces = [];
                         // assumption: one public ip per interface
                         this.subServiceInterfaces.push(this.apiService.getServiceInterfaces(service.shortName, service.version)
                             .subscribe(serviceInterfaces => {
 
                                 safeUnsubscribeList(this.subPublicIps);
+                                this.subPublicIps = [];
```

Comment:
move (together with safeUnsubscribe list) out ouf subscription to capture all ip subscriptions in the same list.